### PR TITLE
GitHub Actions - run linting and specs separately

### PR DESCRIPTION
**Pull request by daaang**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,10 +8,15 @@ on:
 jobs:
   test:
     runs-on: ${{ contains( github.event.pull_request.labels.*.name, 'use_fast_ci') && 'ubuntu-latest-8core' || 'ubuntu-latest' }}
+    strategy:
+      fail-fast: false
```
Comment (daaang) [inline .github/workflows/ci.yml:12]: This is `true` by default and would e.g. stop running tests in the event that linting fails. Setting it to `false` will make tests run to completion, even if linting fails.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,10 +8,15 @@ on:
...
+      matrix:
+        target:
```
Comment (daaang) [inline .github/workflows/ci.yml:14]: The word `target` is arbitrary here, just something to reference later as `${{ matrix.target }}`

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -8,10 +8,15 @@ on:
 jobs:
   test:
     runs-on: ${{ contains( github.event.pull_request.labels.*.name, 'use_fast_ci') && 'ubuntu-latest-8core' || 'ubuntu-latest' }}
```
Comment (rrotter) [inline .github/workflows/ci.yml]: We never want to run lint on ubuntu-latest-8core. We've not used this in a while, and I almost took it out in my PR. We should either remove this now, or modify it so it only applies to "specs" and not "lint".

**Review by rrotter — COMMENTED** [review]
see comment on runner types

File: .github/workflows/ci.yml (hunk 3 above)
Comment (rrotter) [reply .github/workflows/ci.yml]:
```suggestion
    runs-on: ubuntu-latest
```

**Pull request merged**